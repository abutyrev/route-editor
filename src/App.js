import React, { Component } from "react";
import Input from "./components/presentational/Input";
import RouteList from "./components/presentational/RouteList";
import Map from "./components/presentational/Map";

const reorder = (list, startIndex, endIndex) => {
  const result = Array.from(list);
  const [removed] = result.splice(startIndex, 1);
  result.splice(endIndex, 0, removed);

  return result;
};

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      routes: []
    };
    this.id = 0;
  }

  generateId = () => {
    ++this.id;
    return this.id;
  };

  addRoute = route => {
    const routes = this.state.routes.slice(),
      id = this.generateId();

    this.setState({
      routes: [...routes, { id, route }]
    });
  };

  deleteRoute = id => {
    const routes = this.state.routes.filter(route => {
      return route.id !== id;
    });
    this.setState({
      routes
    });
  };

  onDragEnd = result => {
    if (!result.destination) {
      return;
    }

    const routes = reorder(
      this.state.routes,
      result.source.index,
      result.destination.index
    );

    this.setState({
      routes
    });
  };

  render() {
    return (
      <div className="App row">
        <div className="UserField col s4">
          <Input addRoute={this.addRoute} />
          {this.state.routes.length ? (
            <RouteList
              routes={this.state.routes}
              deleteRoute={this.deleteRoute}
              onDragEnd={this.onDragEnd}
            />
          ) : null}
        </div>
        <div className="MapField col s8">
          <Map />
        </div>
      </div>
    );
  }
}

export default App;
