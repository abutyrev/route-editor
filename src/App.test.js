import React from "react";
import { shallow } from "enzyme";

import App from "./App";

describe("App component tests", () => {
  const initialState = {
    points: [],
    isMapReady: false,
    yMap: null
  };
  const AppComponent = shallow(<App />);

  it("should render correctly", () => {
    expect(AppComponent).toMatchSnapshot();
  });

  it("should init with correct state", () => {
    expect(AppComponent.state()).toEqual(initialState);
  });
});
