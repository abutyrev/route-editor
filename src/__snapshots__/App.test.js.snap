// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="UserField col s4"
        >
          <Input
            addPoint={[Function]}
          />
        </div>,
        <div
          className="MapField col s8"
        >
          <MapContainer
            points={Array []}
            updatePoint={[Function]}
          />
        </div>,
      ],
      "className": "App row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input
              addPoint={[Function]}
            />,
            null,
          ],
          "className": "UserField col s4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addPoint": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MapContainer
            points={Array []}
            updatePoint={[Function]}
          />,
          "className": "MapField col s8",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "points": Array [],
            "updatePoint": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="UserField col s4"
          >
            <Input
              addPoint={[Function]}
            />
          </div>,
          <div
            className="MapField col s8"
          >
            <MapContainer
              points={Array []}
              updatePoint={[Function]}
            />
          </div>,
        ],
        "className": "App row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Input
                addPoint={[Function]}
              />,
              null,
            ],
            "className": "UserField col s4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addPoint": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MapContainer
              points={Array []}
              updatePoint={[Function]}
            />,
            "className": "MapField col s8",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "points": Array [],
              "updatePoint": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
